File: app/src/main/java/me/shkschneider/skeleton/SkeletonFragmentActivity.java
package me.shkschneider.skeleton;

import android.os.Bundle;
import android.support.v4.app.Fragment;

import me.shkschneider.app.R;
import me.shkschneider.skeleton.helper.LogHelper;

/**
 * Base FragmentActivity you should use!
 *
 * - void setFragment(Fragment)
 *
 * @see me.shkschneider.skeleton.SkeletonActivity
 */
public class SkeletonFragmentActivity extends SkeletonActivity {

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fragment);
    }

    protected void setFragment(final Fragment fragment) {
        if (fragment == null) {
            LogHelper.warning("Fragment was NULL");
            return ;
        }

        getSupportFragmentManager().beginTransaction()
                .replace(R.id.framelayout, fragment)
                .commit();
    }

}

File: android/src/main/kotlin/me/shkschneider/skeleton/android/core/SkeletonFragmentActivity.kt
package me.shkschneider.skeleton.android.core

import android.os.Bundle
import androidx.fragment.app.Fragment
import me.shkschneider.skeleton.android.R
import me.shkschneider.skeleton.android.core.SkeletonActivity

/**
 * @see SkeletonActivity
 */
abstract class SkeletonFragmentActivity : SkeletonActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.sk_fragmentactivity)
    }

    protected fun setContentFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
                .replace(R.id.frameLayout, fragment)
                .commit()
    }

    override fun onResumeFragments() {
        super.onResumeFragments()
        // fragmentManager.executePendingTransactions()
    }

    override fun onSupportNavigateUp(): Boolean {
        return super.onSupportNavigateUp()
    }

    override fun onBackPressed() {
        if (supportFragmentManager?.backStackEntryCount != 0) {
            supportFragmentManager.popBackStack()
            return
        }
        super.onBackPressed()
    }

}